rest-service/manager_rest/blueprints_manager.py

```diff
@@ -260,7 +336,29 @@ def execute_workflow(self, deployment_id, workflow_id,
 
         return new_execution
 
-    def _execute_system_workflow(self, deployment, wf_id, task_mapping,
+    def _check_for_any_active_executions(self):
+        executions = [
+            e.id
+            for e in self.executions_list(is_include_system_workflows=True)
```

Comment:
`self.executions_list` accepts a filter argument where you can filter by multiple states instead of reading all executions.